GVFS/FastFetch/NativeUnixMethods.cs

```diff
@@ -0,0 +1,146 @@
+                tracer.RelatedError("Error writing file {0}. Win32Exception: {1}", destination, e);
+                throw;
+            }
+            finally
+            {
+                Close(fileDescriptor);
+            }
+        }
+
+        public static bool StatAndUpdateIndexForFile(string path, MemoryMappedViewAccessor indexView, long offset)
+        {
+            NativeStat.StatBuffer st = StatFile(path);
+            Index.IndexEntry indexEntry = new Index.IndexEntry(indexView, offset);
+            indexEntry.MtimeSeconds = (uint)st.MTimespec.Sec;
+            indexEntry.MtimeNanosecondFraction = (uint)st.MTimespec.Nsec;
+            indexEntry.CtimeSeconds = (uint)st.CTimespec.Sec;
+            indexEntry.CtimeNanosecondFraction = (uint)st.CTimespec.Nsec;
+            indexEntry.Size = (uint)st.Size;
+            indexEntry.Dev = (uint)st.Dev;
+            indexEntry.Ino = (uint)st.Ino;
+            indexEntry.Uid = st.UID;
+            indexEntry.Gid = st.GID;
+            return true;
+        }
+
+        [DllImport("libc", EntryPoint = "open", SetLastError = true)]
+        public static extern int Open(string path, int flag, ushort creationMode);
+
+        [DllImport("libc", EntryPoint = "close", SetLastError = true)]
+        public static extern int Close(int fd);
+
+        [DllImport("libc", EntryPoint = "write", SetLastError = true)]
+        private static unsafe extern IntPtr Write(int fileDescriptor, void* buf, IntPtr count);
+
+        private static NativeStat.StatBuffer StatFile(string fileName)
+        {
+            NativeStat.StatBuffer statBuffer = new NativeStat.StatBuffer();
+            if (NativeStat.Stat(fileName, out statBuffer) != 0)
+            {
+                throw new Win32Exception(Marshal.GetLastWin32Error(), $"Failed to stat {fileName}");
+            }
+
+            return statBuffer;
+        }
+
+        private static class NativeStat
+        {
+            // #define  S_IFMT      0170000     /* [XSI] type of file mask */
+            private static readonly ushort IFMT = Convert.ToUInt16("170000", 8);
+
+            // #define  S_IFSOCK    0140000     /* [XSI] socket */
+            private static readonly ushort IFSOCK = Convert.ToUInt16("0140000", 8);
+
+            // #define S_IXUSR     0000100     /* [XSI] X for owner */
+            private static readonly ushort IXUSR = Convert.ToUInt16("100", 8);
+
+            // #define S_IXGRP     0000010     /* [XSI] X for group */
+            private static readonly ushort IXGRP = Convert.ToUInt16("10", 8);
+
+            // #define S_IXOTH     0000001     /* [XSI] X for other */
+            private static readonly ushort IXOTH = Convert.ToUInt16("1", 8);
+
+            public static bool IsSock(ushort mode)
+            {
+                // #define  S_ISSOCK(m) (((m) & S_IFMT) == S_IFSOCK)    /* socket */
+                return (mode & IFMT) == IFSOCK;
+            }
+
+            public static bool IsExecutable(ushort mode)

```

Comment:
If `FastFetch` is not using any of the readonly ushorts (e.g, `IXGRP`) or helper methods (`IsSock`, `IsExecutable`) they can be removed.